Yank/multistate/sams.py

```diff
@@ -0,0 +1,419 @@
+            if scheme not in supported_schemes:
+                raise ValueError("Unknown update scheme '{}'. Supported values "
+                                 "are {}.".format(scheme, supported_schemes))
+            return scheme
+
+        @staticmethod
+        def _weight_update_method_validator(instance, scheme):
+            supported_schemes = ['optimal', 'rao-blackwellized']
+            if scheme not in supported_schemes:
+                raise ValueError("Unknown update scheme '{}'. Supported values "
+                                 "are {}.".format(scheme, supported_schemes))
+            return scheme
+
+        @staticmethod
+        def _adapt_target_probabilities_validator(instance, scheme):
+            supported_schemes = [True, False]
+            if scheme not in supported_schemes:
+                raise ValueError("Unknown update scheme '{}'. Supported values "
+                                 "are {}.".format(scheme, supported_schemes))
+            return scheme
+
+    log_target_probabilities = _StoredProperty('log_target_probabilities', validate_function=None)
+    state_update_scheme = _StoredProperty('state_update_scheme', validate_function=_StoredProperty._state_update_scheme_validator)
+    locality = _StoredProperty('locality', validate_function=None)
+    update_stages = _StoredProperty('update_stages', validate_function=_StoredProperty._update_stages_validator)
+    flatness_threshold = _StoredProperty('flatness_threshold', validate_function=None)
+    weight_update_method = _StoredProperty('weight_update_method', validate_function=_StoredProperty._weight_update_method_validator)
+    adapt_target_probabilities = _StoredProperty('adapt_target_probabilities', validate_function=_StoredProperty._adapt_target_probabilities_validator)
+
+    def create(self, thermodynamic_states: list, sampler_states:list, storage,
+               **kwargs):
+        """Initialize SAMS sampler.
+
+        Parameters
+        ----------
+        thermodynamic_states : list of openmmtools.states.ThermodynamicState
+            Thermodynamic states to simulate, where one replica is allocated per state.
+            Each state must have a system with the same number of atoms.
+        sampler_states : list of openmmtools.states.SamplerState
+            One or more sets of initial sampler states.
+            The number of replicas is determined by the number of sampler states provided,
+            and does not need to match the number of thermodynamic states.
+            Most commonly, a single sampler state is provided.
+        storage : str or Reporter
+            If str: path to the storage file, checkpoint options are default
+            If Reporter: Instanced :class:`Reporter` class, checkpoint information is read from
+            In the future this will be able to take a Storage class as well.
+        initial_thermodynamic_states : None or list or array-like of int of length len(sampler_states), optional,
```

Comment:
Since this is SAMS, should there be only 1 sampler state? Or will SAMS work with multiple sampler states, each running independently?

Reply:
SAMS will use as many replicas as you provide initial SamplerStates. It should converge at a rate proportional to the number of replicas (assuming no slow conformational degrees of freedom).